lib/guest_installation_and_configuration_base.pm

```diff
@@ -911,32 +927,87 @@ EOF
         if (!($self->{guest_installation_media} =~ /-$self->{guest_version}-/im)) {
             record_info("Guest $self->{guest_name} installation media $self->{guest_installation_media} does not match with version $self->{guest_version}", "Going to correct it !");
-            $self->{guest_installation_media} =~ /-(\d+)-(sp(\d)*){0,1}/im;
+            my $_guest_version_major_indicator = ($self->{guest_os_name} =~ /sles/im ? ''   : 'R');
+            my $_guest_version_minor_indicator = ($self->{guest_os_name} =~ /sles/im ? 'SP' : 'U');
+            $self->{guest_installation_media} =~ /-((r)?(\d*))-((sp|u)?(\d*))?/im;
             if ($self->{guest_version_minor} ne 0) {
-                $self->{guest_installation_media} =~ s/$1-$2/$self->{guest_version_major}-$self->{guest_version_minor}/im if ($2 ne '');
-                $self->{guest_installation_media} =~ s/$1/$self->{guest_version_major}-SP$self->{guest_version_minor}/im  if ($2 eq '');
+                if ($4 ne '') {
+                    $self->{guest_installation_media} =~ s/-$1-$4/-${_guest_version_major_indicator}$self->{guest_version_major}-${_guest_version_minor_indicator}$self->{guest_version_minor}/im;
+                }
+                else {
+                    $self->{guest_installation_media} =~ s/-$1/-${_guest_version_major_indicator}$self->{guest_version_major}-${_guest_version_minor_indicator}$self->{guest_version_minor}/im;
+                }
             }
             else {
-                $self->{guest_installation_media} =~ s/$1-$2/$self->{guest_version_major}/im if ($2 ne '');
-                $self->{guest_installation_media} =~ s/$1/$self->{guest_version_major}/im    if ($2 eq '');
+                if ($4 ne '') {
+                    $self->{guest_installation_media} =~ s/-$1-$4/-${_guest_version_major_indicator}$self->{guest_version_major}/im;
+                }
+                else {
+                    $self->{guest_installation_media} =~ s/-$1/-${_guest_version_major_indicator}$self->{guest_version_major}/im;
+                }
             }
         }
-        record_info("Guest $self->{guest_name} is going to use installation media $self->{guest_installation_media}", "Please check it out !");
-        $self->{guest_installation_method_options} = "--location $self->{guest_installation_media}";
     }
-    $self->{guest_installation_method_options} = $self->{guest_installation_method_options} . ($self->{guest_installation_fine_grained} ne '' ? " --install $self->{guest_installation_fine_grained}" : '') if ($self->{guest_installation_fine_grained} ne '');
-    $self->{guest_installation_method_options} = $self->{guest_installation_method_options} . ($self->{guest_autoconsole} ne '' ? " --autoconsole $self->{guest_autoconsole}" : ' --noautoconsole');
+
+#If guest chooses to use iso installation media, then this iso media should be available on INSTALLATION_MEDIA_NFS_SHARE and mounted locally at INSTALLATION_MEDIA_LOCAL_SHARE.
```

Comment:
same here

Reply:
CI tells me to do this way otherwise it will not let it pass. I really can not change as you suggested here.

Reply:
I tried it again. No way. Keep it the current way unless CI changes.